e2e/explorer/api/src/app/etherscan/logs.ts

```diff
@@ -0,0 +1,41 @@
+import { GetParams, Logs } from "./types";
+import { store } from '../store';
+
+export function getLogs(params: GetParams<Logs>) {
+  const { fromBlock, toBlock, address } = params;
+  if (!fromBlock || !toBlock || !address) throw new Error('Error! Missing or invalid Action name');
+
+  let filter: {from: number; to: number; operator: 'or' | 'and' };
+  from:
+  for (let i = 0; i<3; i++) {
+    for (let j = i +1; j<4; j++) {
+      const operator = params[`topic${i}_${j}_opr` as any];
+      if(operator) {
+        filter = { from: i, to: j, operator};
+        break from;
+      }
+    }
+  };
+  if (!filter) {
+    filter = { from: 0, to: 3, operator: "and" };
+  }
+
+  return Object.values(store.logs).flat().filter(log => {
+    if (log.address !== address) return false;
+    if ((typeof fromBlock === "string" && fromBlock !== "latest") || (typeof toBlock === "string" && toBlock !== "latest")) return false;
+    if (log.blockNumber <= fromBlock || log.blockNumber >= toBlock ) return false;
```

Comment:
I think you can take the equality as a good value
```suggestion
    if (log.blockNumber < fromBlock || log.blockNumber > toBlock ) return false;
```